pkg/util/push/otlp.go

```diff
@@ -0,0 +1,184 @@
+package push
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote/otlptranslator/prometheusremotewrite"
+	"github.com/weaveworks/common/httpgrpc"
+	"github.com/weaveworks/common/middleware"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp"
+
+	"github.com/cortexproject/cortex/pkg/cortexpb"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/log"
+)
+
+// OTLPHandler is a http.Handler which accepts OTLP metrics.
+func OTLPHandler(sourceIPs *middleware.SourceIPExtractor, push Func) http.Handler {
```

Comment:
Can we try to utilize https://github.com/prometheus/prometheus/blob/main/storage/remote/write_handler.go#L204 as much as possible?

For example, can `DecodeOTLPWriteRequest` be used here?

Reply:
done